### DISCUSSION TITLE
yyyyyyyyNOCN-53 failed:yyy [INFO] TEST FAIL. ```

### DISCUSSION THREAD

**Discussion by hao-888** [kkkgo/PaoPaoDNS#90] [问答和帮助]
-> test start `1704266441`
```rust
yyyyyyyyNOCN-53 failed:yyy
[INFO] TEST FAIL.
```
-> test end `1704266445`

什么情况有点懵逼

我的docker配置
docker run -d \
--name paopaodns \
-v /home/mydata:/data \
-e UPDATE=weekly \
-e CNAUTO=yes \
-e CNFALL=yes \
-e CN_TRACKER=yes \
-e IPV6=no \
-e CUSTOM_FORWARD=10.0.0.250:53  \
-e  AUTO_FORWARD=yes   \
-e  AUTO_FORWARD_CHECK=yes \
-e  USE_MARK_DATA=yes \
-e  CN_TRACKER=yes \
-e  ADDINFO=yes \
-e  SHUFFLE=yes \
-e  RULES_TTL=604800 \
-e  TZ=Asia/Shanghai \
--restart unless-stopped \
-p 53:53/tcp  \
-p 53:53/udp  \
-p 5304:5304/tcp  \
-p 5304:5304/udp  \
-p 7889:7889/tcp  \
-p 7889:7889/udp  \
sliamb/paopaodns

**Comment by hao-888** [comment]
64 bytes from 223.5.5.5: seq=0 ttl=117 time=11.515 ms

--- 223.5.5.5 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 11.515/11.515/11.515 ms
PING 119.29.29.29 (119.29.29.29): 56 data bytes
64 bytes from 119.29.29.29: seq=0 ttl=52 time=17.194 ms

--- 119.29.29.29 ping statistics ---
1 packets transmitted, 1 packets received, 0% packet loss
round-trip min/avg/max = 17.194/17.194/17.194 ms
Server:         223.5.5.5
Address:        223.5.5.5#53

Non-authoritative answer:
www.taobao.com  canonical name = www.taobao.com.danuoyi.tbcache.com.
Name:   www.taobao.com.danuoyi.tbcache.com
Address: 119.188.121.193
Name:   www.taobao.com.danuoyi.tbcache.com
Address: 119.188.121.192
Name:   www.taobao.com.danuoyi.tbcache.com
Address: 2408:8719:64:53:3::3da
Name:   www.taobao.com.danuoyi.tbcache.com
Address: 2408:8719:64:53:3::3db

Server:         119.29.29.29
Address:        119.29.29.29#53

Non-authoritative answer:
www.qq.com      canonical name = ins-r23tsuuf.ias.tencent-cloud.net.
Name:   ins-r23tsuuf.ias.tencent-cloud.net
Address: 221.198.70.47
Name:   ins-r23tsuuf.ias.tencent-cloud.net
Address: 2408:8711:10:1002::19
Name:   ins-r23tsuuf.ias.tencent-cloud.net
Address: 2408:8711:10:1003::30

*********************************************************************************

[INFO] ENV
*********************************************************************************

====ENV TEST====
[OK]DATA_writeable-
[OK]DATA_readable-
MEM:220m 450m 500000 750mb
prefPC:68
CORES:-2-
POWCORES:-2-
ulimit :-1048576-
FDLIM :-4096-
TZ:-Asia/Shanghai-
UPDATE:-weekly-
DNS_SERVERNAME:-PaoPaoDNS,blog.03k.org-
SERVER_IP:-none-
ETHIP:-172.17.0.2-
DNSPORT:-53-
SOCKS5:-IP:PORT-
CNAUTO:-yes-
IPV6:-no-
CNFALL:-yes-
CUSTOM_FORWARD:-10.0.0.250:53-
AUTO_FORWARD:-yes-
AUTO_FORWARD_CHECK:-yes-
USE_MARK_DATA:-yes-
RULES_TTL:-604800-
CUSTOM_FORWARD_TTL:-0-
SHUFFLE:-yes-
CN_TRACKER:-yes-
USE_HOSTS:-no-
HTTP_FILE:-no-
SAFEMODE:--
QUERY_TIME:-2000ms-
ADDINFO:-yes-
PLATFORM:-Linux 6c2e0177c980 6.5.11-4-pve #1 SMP PREEMPT_DYNAMIC PMX 6.5.11-4 (2023-11-20T10:19Z) x86_64 Linux-
====ENV TEST====
mosdns kkkgo/mosdns:231210.1
*********************************************************************************

[INFO] PS
*********************************************************************************

PID   USER     TIME  COMMAND
    1 root      0:00 {init.sh} /bin/sh /usr/sbin/init.sh
   23 root      0:00 crond
   55 root      0:01 redis-server unixsocket:/tmp/redis.sock
  286 root      0:00 dnscrypt-proxy -config /data/dnscrypt-resolvers/dnscrypt.toml
  312 root      0:00 unbound -c /tmp/unbound_forward.conf -p
  315 root      0:00 mosdns start -d /tmp -c /tmp/mosdns.yaml
  325 root      0:00 {watch_list.sh} /bin/sh /usr/sbin/watch_list.sh
  335 root      0:00 tail -f /dev/null
  336 root      0:00 unbound -c /tmp/unbound_raw.conf -p
  367 root      0:00 inotifywait -e modify,delete /etc/unbound/named.cache /data/Country-only-cn-private.mmdb /data/force_cn_list.txt /data/force_nocn_list.txt /data/custom_env.ini /data/global_mark.dat /data/trackerslist.txt /data/force_forward_list.txt /data/force_ttl_rules.txt
  821 root      0:00 {debug.sh} /bin/sh /usr/sbin/debug.sh
  841 root      0:00 ps -ef
*********************************************************************************

[INFO] TOP
*********************************************************************************

CPU:   2% usr   0% sys   0% nic  94% idle   0% io   0% irq   2% sirq
  PID  PPID USER     STAT   VSZ %VSZ CPU %CPU COMMAND
  336     1 root     S    1219m   7%   0   3% unbound -c /tmp/unbound_raw.conf -
  315     1 root     S    1211m   7%   3   0% mosdns start -d /tmp -c /tmp/mosdn
  286     1 root     S    1211m   7%   1   0% dnscrypt-proxy -config /data/dnscr
   55     1 root     S    22120   0%   3   0% redis-server unixsocket:/tmp/redis
  312     1 root     S    20236   0%   1   0% unbound -c /tmp/unbound_forward.co
  325     1 root     S     1708   0%   3   0% {watch_list.sh} /bin/sh /usr/sbin/
    1     0 root     S     1632   0%   3   0% {init.sh} /bin/sh /usr/sbin/init.s
  821     0 root     S     1628   0%   1   0% {debug.sh} /bin/sh /usr/sbin/debug
  842   821 root     R     1624   0%   0   0% top -n1
  335     1 root     S     1616   0%   2   0% tail -f /dev/null
  843   821 root     S     1612   0%   3   0% grep %
  367   325 root     S     1072   0%   2   0% inotifywait -e modify,delete /etc/
   23     1 root     S      860   0%   1   0% crond
*********************************************************************************

[INFO] REDIS
*********************************************************************************

used_memory_human:962.16K
used_memory_rss_human:3.72M
used_memory_peak_human:975.48K
total_system_memory_human:15.45G
used_memory_lua_human:31.00K
used_memory_vm_total_human:63.00K
used_memory_scripts_human:181B
maxmemory_human:750.00M
7
*********************************************************************************

[TEST] IP ROUTE
*********************************************************************************

CN IP URL:
222.132.201.119
-
222.132.201.119
--
222.132.201.119
CN RAW-IP URL:
222.132.201.119
------------------
Non-CN IP URL:
222.132.201.119
-
222.132.201.119
--
222.132.201.119
Non-CN RAW-IP URL:
222.132.201.119
-
222.132.201.119
--
222.132.201.119
---
222.132.201.119
------------------
IP INFO:
222.132.201.119
CN,Linyi,Shandong
ASN4837/China Unicom
HTTP/1.1 
Mozilla/5.0 Gecko/20100101 Firefox/120.0 https://github.com/kkkgo/PaoPaoDNS
Asia/Shanghai Time: 1/3/2024, 3:22:20 PM
[INFO] force_cn_list
domain:whoami.ds.akahelp.net
domain:whoami.03k.org
MOSDNS WHOAMI :
akahelp: "ns" "222.132.201.119"
03k: 222.132.201.119
UNBOUND WHOAMI:
akahelp: "ns" "222.132.201.119"
03k: 222.132.201.119
*********************************************************************************

[TEST] HIJACK
*********************************************************************************

;; communications error to 9.8.7.5#53: timed out
;; no servers could be reached

;; communications error to 9.8.7.6#53: timed out
;; no servers could be reached

HIJACK 127.0.0.1 = 58.217.249.139
*********************************************************************************

[TEST] DIG-CN [taobao]
*********************************************************************************

MOSDNS CN:
www.taobao.com.danuoyi.tbcache.com.
119.188.121.192
119.188.121.193
UNBOUND CN:
www.taobao.com.danuoyi.tbcache.com.
119.188.121.193
119.188.121.192
[TEST] DIG-NOCN [youtube]
MOSDNS NOCN:
DNSCRYPT-UNBOUND NOCN:
youtube-ui.l.google.com.
142.250.66.142
142.250.207.78
DNSCRYPT NOCN:
youtube-ui.l.google.com.
142.250.185.142
142.250.185.174
DNSCRYPT-SOCKS5 NOCN:
;; communications error to 127.0.0.1#5303: connection refused
;; no servers could be reached

*********************************************************************************

CUSTOM_FORWARD TEST [youtube]:
;; communications error to 10.0.0.250#53: timed out

**Comment by kkkgo** [comment]
;; communications error to 10.0.0.250#53: timed out

你自定义转发的DNS服务器失败了，不通。
